### PR TITLE
[BE-198] Set different exit status for aborted builds

### PR DESCRIPTION
**Pull request by lpusok**
### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-io/bitrise/blob/master/.github/CONTRIBUTING.md)
- [x] `README.md` is updated with the changes (if needed)

### Version

Requires a *PATCH* [version update](https://semver.org/)

### Context

Set different exit codes based if there were Steps which were aborted:

- Step aborted by custom timeout set in the bitrise.yml: 11
- Step aborted by no output timeout: 12
- Any other failure: 1

These will be used to display a relevant message on the frontend.

Resolves: https://bitrise.atlassian.net/browse/BE-198

### CODE REVIEW

File: cli/run.go [hunk 1]
```diff
@@ -122,17 +122,35 @@ func runAndExit(bitriseConfig models.BitriseDataModel, inventoryEnvironments []e
...
 	os.Exit(0)
 }
 
+func getExitCodeFromFailedBuild(buildRunResults models.BuildRunResultsModel) int {
+	if !buildRunResults.IsBuildFailed() {
+		return 0
+	}
+
+	if buildRunResults.IsBuildAbortedWithNoOutputTimeout() {
+		return 12
```
Comment (matrangam) [inline cli/run.go:144]: Is it reasonable to make some constants for the return values?

File: cli/run.go [hunk 2]
```diff
@@ -122,17 +122,35 @@ func runAndExit(bitriseConfig models.BitriseDataModel, inventoryEnvironments []e
...
+	}
+
+	if buildRunResults.IsBuildAbortedWithTimeout() {
+		return 11
```
Comment (matrangam) [inline cli/run.go:148]: Is it reasonable to make some constants for the return values?

**Review by matrangam — COMMENTED** [review]
Is it reasonable to make some constants for the return values?

File: models/models_methods.go [hunk 3]
```diff
@@ -1166,34 +1166,49 @@ func (sIDData StepIDData) IsUniqueResourceID() bool {
...
 
-// ResultsCount ...
 func (buildRes BuildRunResultsModel) ResultsCount() int {
 	return len(buildRes.SuccessSteps) + len(buildRes.FailedSteps) + len(buildRes.FailedSkippableSteps) + len(buildRes.SkippedSteps)
 }
 
+func (buildRes BuildRunResultsModel) IsBuildAbortedWithTimeout() bool {
+	for _, stepResult := range buildRes.FailedSteps {
+		if stepResult.Status == StepRunStatusAbortedTimeout {
+			return true
```
Comment (ofalvai) [inline models/models_methods.go:1188]: Is this true though? What happens if a step that is marked as `is_skippable` is aborted this way? In this case the workflow will still be successful, no?

Comment (lpusok) [reply models/models_methods.go:1188]: Failed but skipped steps are collected separately: https://github.com/bitrise-io/bitrise/blob/9127b0e96e9392257ff85e57b8dd267fd2ca4506/models/models.go#L141

**Review by matrangam — DISMISSED** [review]

**Review by ofalvai — APPROVED** [review]

**Review by ofalvai — APPROVED** [review]

**Pull request merged**